src/AvPortalClient.php

```diff
@@ -53,6 +75,14 @@ public function query(array $options = []): ?array {
       'pagesize' => 15,
     ];
 
+    // Generate a cache ID that takes into consideration all the query
+    // parameters.
+    $cid = 'media_avportal:client:query:' . serialize($options);
```

Comment:
Is it safe to use cid with serialized options. Cid field in database have limit of 255 chars. potentially we could have fatal error on try to set cache (especially for searches where field for search could have very long string size).

Reply:
The cache `cid` is automatically truncated by the database implementation.
See `\Drupal\Core\Cache\DatabaseBackend::normalizeCid()`.